addons/pos_loyalty/static/src/js/ProductScreen.js

```diff
@@ -56,7 +56,19 @@ export const PosLoyaltyProductScreen = (ProductScreen) =>
                     return false;
                 }
                 const trimmedCode = code.trim();
-                if (trimmedCode && trimmedCode.startsWith('044')) {
+                const fetchedNomenclatureCodePatterns = await this.rpc({
+                    model: 'barcode.rule',
+                    method: 'search_read',
+                    args: [
+                        [['type', '=', 'coupon'], ['barcode_nomenclature_id', '=', this.env.pos.company.nomenclature_id[0]]],
+                        ['pattern'],
+                    ],
+                });
+                const nomenclatureCodePatterns = fetchedNomenclatureCodePatterns[0].pattern.split("|")
+                let trimmedCodeValid = _.find(nomenclatureCodePatterns, function(pattern) {
```

Comment:
```suggestion
                const trimmedCodeValid = _.find(nomenclatureCodePatterns, function(pattern) {
```
No need to make this a `let` variable if we're not going to reassign it later.